qiskit/primitives/primitive_job.py

```diff
@@ -27,14 +27,16 @@ class PrimitiveJob(BasePrimitiveJob[ResultT, JobStatus]):
     Primitive job class for the reference implementations of Primitives.
     """
 
-    def __init__(self, function, *args, **kwargs):
+    def __init__(self, *args, function=None, **kwargs):
```

Comment:
This does not work as expected because `function` is always included in `args` and `self._function` will be always `None`.

Reply:
Perhaps you want something like the following?
```suggestion
    def __init__(self, function=None, *args, **kwargs):
```

Reply:
I had that originally but got a pylint error, I think we'll have to ignore `keyword-arg-before-vararg`